src/compile/data/dataflow.ts

```diff
@@ -127,3 +135,23 @@ export interface DependentNode {
    */
   dependentFields: () => StringSet;
 }
+
+
+/**
+ * Trait for nodes that can be cloned.
+ */
+export interface ClonableNode {
+  /**
+   * Clone this node with a deep copy.
+   */
+  clone: () => this;
+}
+
+
+export function isNewFieldNode(node: any): node is NewFieldNode {
+  return 'produces' in node && isFunction(node.produces);
+}
+
+export function isClonable(node: any): node is ClonableNode {
```

Comment:
Instead of using `ClonableNode` mixins and `isClonable` function like this.

I think we could just add two methods `isClonable()` and `clone()` to all DataFlowNode. 
`clone()` by default (if not overridden) can throw `new Error("Not clonable")`.